fastembed/rerank/cross_encoder/onnx_text_cross_encoder.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Union, Iterable, Dict, Any, Sequence, Optional
+from fastembed.common import OnnxProvider
+from fastembed.common.onnx_model import OnnxOutputContext
+from fastembed.rerank.cross_encoder.onnx_text_model import OnnxCrossEncoderModel
+from fastembed.rerank.cross_encoder.text_cross_encoder_base import TextCrossEncoderBase
+import numpy as np
+
+supported_onnx_models = [
+    {
+        "model": "Xenova/ms-marco-MiniLM-L-6-v2",
+        "dim": 1024,
+        "sources": {
+            "hf": "Xenova/ms-marco-MiniLM-L-6-v2",
+        },
+        "model_file": "onnx/model.onnx"
+    },
+    {
+        "model": "Xenova/ms-marco-MiniLM-L-12-v2",
+        "dim": 1024,
+        "sources": {
+            "hf": "Xenova/ms-marco-MiniLM-L-12-v2",
+        },
+        "model_file": "onnx/model.onnx"
+    },
+    {
+        "model": "BAAI/bge-reranker-base",
+        "dim": 1024,
+        "sources": {
+            "hf": "BAAI/bge-reranker-base",
+        },
+        "model_file": "onnx/model.onnx"
+    }
+]
+
+class OnnxTextCrossEncoder(TextCrossEncoderBase, OnnxCrossEncoderModel):
+    ONNX_OUTPUT_NAMES = ["logits"]
+
+    @classmethod
+    def list_supported_models(cls) -> List[Dict[str, Any]]:
+        return supported_onnx_models
+
+    def __init__(
+            self, 
+            model_name: str, 
+            cache_dir: Optional[str] = None, 
+            threads: Optional[int] = None, 
+            providers: Optional[Sequence[OnnxProvider]] = None, 
+            **kwargs,
+    ):
+        TextCrossEncoderBase.__init__(self, model_name, cache_dir, threads, **kwargs)
+        OnnxCrossEncoderModel.__init__(self, model_name, cache_dir, threads, providers, **kwargs)
```

Comment:
We usually do not call OnnxCrossEncoderModel init, can we avoid calling it?

Reply:
I think the whole init method can probably be copied from here

https://github.com/qdrant/fastembed/blob/0e258ab875c377c8b7961f2e37afd16cafdaa801/fastembed/text/onnx_embedding.py#L189C9-L202C10